File: longestSWMBA.cpp
#include <vector>

class Solution {
public:
    int longestSubarray(std::vector<int>& nums) {
        int maxVal = 0, maxLen = 0, currLen = 0;

        for (int num : nums) {
            if (num > maxVal) {
                maxVal = num;
                maxLen = currLen = 1;
            } else if (num == maxVal) {
                ++currLen;
                if (currLen > maxLen) maxLen = currLen;
            } else {
                currLen = 0;
            }
        }
        return maxLen;
    }
};

File: LongestSWMBA.py
from typing import List

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        max_val = float('-inf')
        longest = curr_len = 0

        for num in nums:
            if num > max_val:
                max_val = num
                curr_len = 1
                longest = 1
            elif num == max_val:
                curr_len += 1
                longest = max(longest, curr_len)
            else:
                curr_len = 0

        return longest
